### PR TITLE
Improve `subl` completions for Sublime Text v4

### PR DESCRIPTION
**Pull request by stdedos**
Additionally, add model lines

Signed-off-by: Stavros Ntentos <133706+stdedos@users.noreply.github.com>

### CODE REVIEW

File: Completion/X/Command/_sublimetext
Comment (phy1729) [inline Completion/X/Command/_sublimetext:10]: [Sublime's docs][1] contain "The available flags will vary per operating system," so I'm hesitant to remove any flags without an ok from @okdana (who added the flag).

[1]: https://www.sublimetext.com/docs/command_line.html

File: Completion/X/Command/_sublimetext
Comment (phy1729) [inline Completion/X/Command/_sublimetext:10]: The other specs are in the form `'(-x --xlong-form)'{-x.--xlong-form}'[description]'` because there is a short and long form for the same setting. Since `--safe-mode` only has a long option, `'--safe-mode[Launch using a sandboxed (clean) environment]'` suffices. Same for `--launch-or-new-window`.

File: Completion/X/Command/_sublimetext
Comment (phy1729) [inline Completion/X/Command/_sublimetext:22]: As most completers don't have mode lines, I think this part should be left out.

File: Completion/X/Command/_sublimetext
Comment (stdedos) [reply Completion/X/Command/_sublimetext:10]: Would've been nice to also conditionally output flag - but at least Sublime does not seem to do anything either way on Ubuntu.

I can revert since you hesitate.

File: Completion/X/Command/_sublimetext
Comment (stdedos) [reply Completion/X/Command/_sublimetext:10]: I have no idea how the syntax works (and https://github.com/vapniks/zsh-completions/blob/master/zsh-completions-howto.org, https://zsh.sourceforge.io/Doc/Release/Completion-System.html#:~:text=grouping%20options don't seem to explain it adequately to my level of understanding).

File: Completion/X/Command/_sublimetext
Comment (danielshahaf) [reply Completion/X/Command/_sublimetext:10]: Conditional behaviour is possible. What would the condition be?

File: Completion/X/Command/_sublimetext
Comment (danielshahaf) [reply Completion/X/Command/_sublimetext]: In the second link, search for "Each of the forms above may be preceded by a list in parentheses of option names and argument numbers". The grouping facility is unrelated and rarely used.

File: Completion/X/Command/_sublimetext
Comment (stdedos) [reply Completion/X/Command/_sublimetext:10]: All arguments must exist in `subl --help` (by means of grepping??), otherwise not offered

I know it's not a panacea convention, but a lot of respectful utils offer their arguments on either `-h`/`--help` - Sublime does too

Comment (okdana) [reply Completion/X/Command/_sublimetext:10]: `-s`/`--stay` is still there on macOS at least in `subl` 4113. If that option is platform-specific, like Daniel said, we could easily make it conditional. Determining the applicability of options by looking at `--help` output is possible, but not common; in this case  i think it'd be fine to just look at `$OSTYPE`

File: Completion/X/Command/_sublimetext
Comment (okdana) [inline Completion/X/Command/_sublimetext:11]: Completion descriptions should not use sentence case (they should start with a lower-case letter). And since the existing descriptions use 'telegraph' style i would make these match (omit 'a' and 'the' where possible)

**Review by okdana — CHANGES_REQUESTED** [review]

File: Completion/X/Command/_sublimetext
Comment (stdedos) [reply Completion/X/Command/_sublimetext:11]: I disagree, but whatever - if those are the rules ...

File: Completion/X/Command/_sublimetext
Comment (stdedos) [reply Completion/X/Command/_sublimetext:10]: > in this case i think it'd be fine to just look at `$OSTYPE`

"As far as you know" it's based on `$OSTYPE`. Maybe it's only on MacOS? Or it's only not on Linux? 😕
Blackbox approach (e.g. grepping) makes less assumptions on things we don't know for sure.

However, I am not going to tackle it anyway - especially now that you verified it's still a thing.

File: Completion/X/Command/_sublimetext
Comment (okdana) [reply Completion/X/Command/_sublimetext]: The no-sentence-case thing is mentioned in [the style guide](https://github.com/zsh-users/zsh/blob/master/Etc/completion-style-guide) (under *Descriptions*), and most other functions shipped with zsh follow it. I'm not sure if we have any specific guidance re: telegraph style, but it's certainly common, and if nothing else it's usually a good idea to match a file's existing style

Comment (stdedos) [reply Completion/X/Command/_sublimetext]: > The no-sentence-case thing is mentioned in the style guide (under Descriptions), and most other functions shipped with zsh follow it. ...

Yeah, I got that - hence the new commit.

However, it sounds so ... weird.
And technically, also on the same page it mentions:
> It is a good idea to copy descriptions from the command's man page or --help output.

File: Completion/X/Command/_sublimetext
Comment (okdana) [reply Completion/X/Command/_sublimetext:10]: > Maybe it's only on MacOS? Or it's only not on Linux?

Either/both of those conditions could be handled by checking `$OSTYPE`, and it's how most other functions with platform-specific options do it. But yeah i haven't actually checked to see which options are available on which platforms

> However, I am not going to tackle it anyway

That's fine, we can revisit later if we want

**Review by okdana — APPROVED** [review]
Looks OK, thank you. I will merge later (unless someone else wants to do it)

File: Completion/X/Command/_sublimetext
Comment (danielshahaf) [reply Completion/X/Command/_sublimetext]: That just advises not to reinvent the wheel when describing options. Some degree of editing is usually necessary, e.g., when the --help output of an option that takes an argument names the argument.

**Comment by okdana** [comment]
Sorry for the delay. Merged in 765bc1470

**Pull request closed without merging**
